components/brave_wallet_ui/common/async/base-query-cache.ts

```diff
@@ -410,23 +410,47 @@ export class BaseQueryCache {
     if (!this._nftMetadataRegistry[tokenId]) {
       const { jsonRpcService } = getAPIProxy()
 
-      const result =
-        tokenArg.coin === BraveWallet.CoinType.ETH
-          ? tokenArg.isErc721
-            ? await jsonRpcService.getERC721Metadata(
-                tokenArg.contractAddress,
-                tokenArg.tokenId,
-                tokenArg.chainId
-              )
-            : await jsonRpcService.getERC1155Metadata(
-                tokenArg.contractAddress,
-                tokenArg.tokenId,
-                tokenArg.chainId
-              )
-          : await jsonRpcService.getSolTokenMetadata(
-              tokenArg.chainId,
-              tokenArg.contractAddress
-            )
+      let result;
+      if (tokenArg.isCompressed) {
+        const compressedMetadata = await jsonRpcService.getNftMetadatas(
+          tokenArg.coin,
+          [tokenArg.chainId],
+          [tokenArg.contractAddress],
+          [tokenArg.tokenId]
+        );
+
+        let metadata = JSON.stringify({
+          name: compressedMetadata.metadatas[0].name,
```

Comment:
The API accepts a list because it operates on lists elsewhere, in this I just needed to get the metadata for one NFT.  I actually removed this code though since it was for debug purposes.